### PR TITLE
Fix redirect to tab when there is no tab explicitly selected

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

There are [some issues](https://github.com/sonata-project/SonataAdminBundle/pull/5314#issuecomment-477323631) after https://github.com/sonata-project/SonataAdminBundle/pull/5314.

I am targeting this branch, because {reason}.

## Changelog

```markdown
### Fixed
- Fixed redirecting to a blank tab after saving an object
- Fixed modifying form values changing tabs
```

### CODE REVIEW

File: src/Controller/CRUDController.php [hunk 1]
```diff
@@ -1270,7 +1270,7 @@ protected function redirectTo($object)
                     $url = $this->admin->generateObjectUrl(
                         $route,
                         $object,
-                        $request->request->has('_tab') ? ['_tab' => $request->request->get('_tab')] : []
+                        '' !== $request->request->get('_tab', '') ? ['_tab' => $request->request->get('_tab')] : []
```
Comment (greg0ire) [inline src/Controller/CRUDController.php]: There is a shorter solution:
```suggestion
                        array_filter(['_tab' => $request->request->get('_tab')])
```

**Comment by greg0ire** [comment]
If you fix several issues, please make one commit per issue.

File: src/Controller/CRUDController.php [hunk 2]
```diff
@@ -1270,7 +1270,7 @@ protected function redirectTo($object)
                     $url = $this->admin->generateObjectUrl(
                         $route,
                         $object,
-                        $request->request->has('_tab') ? ['_tab' => $request->request->get('_tab')] : []
```
Comment (core23) [inline src/Controller/CRUDController.php]: Please create an own method for this

**Comment by franmomu** [comment]
Sorry I removed the repository 🤦‍♂️, so I created a new [PR](https://github.com/sonata-project/SonataAdminBundle/pull/5512)

**Pull request closed without merging**